src/ralph/assets/models/networks.py

```diff
@@ -0,0 +1,389 @@
+        Returns:
+            True if networks is private, False if is public
+        """
+        ip = ipaddr.IPAddress(self.address.split('/')[0])
+        return (
+            ip in ipaddr.IPNetwork('10.0.0.0/8') or
+            ip in ipaddr.IPNetwork('172.16.0.0/12') or
+            ip in ipaddr.IPNetwork('192.168.0.0/16')
+        )
+
+    def get_subnetworks(self, networks=None):
+        """
+        Return list of all L3 subnetworks this network contains.
+        Only first level of children networks are returned.
+        """
+        if not networks:
+            networks = Network.objects.filter(
+                # TODO
+                data_center=self.data_center,
+                min_ip__gte=self.min_ip,
+                max_ip__lte=self.max_ip,
+            ).exclude(
+                pk=self.id,
+            ).order_by('-min_ip', 'max_ip')
+        subnets = sorted(list(networks), key=lambda net: net.get_netmask())
+        new_subnets = list(subnets)
+        for net, netw in enumerate(subnets):
+            net_address = ipaddr.IPNetwork(netw.address)
+            for i, sub in enumerate(subnets):
+                sub_addr = ipaddr.IPNetwork(sub.address)
+                if sub_addr != net_address and sub_addr in net_address:
+                    if sub in new_subnets:
+                        new_subnets.remove(sub)
+        new_subnets = sorted(new_subnets, key=lambda net: net.min_ip)
+        return new_subnets
+
+
+@python_2_unicode_compatible
+class Network(NamedMixin, TimeStampMixin, AbstractNetwork, models.Model):
+    """
+    """
+    class Meta:
+        verbose_name = _('network')
+        verbose_name_plural = _('networks')
+        ordering = ('vlan',)
+
+    def __str__(self):
+        return '{} ({})'.format(self.name, self.address)
+
+
+class NetworkTerminator(NamedMixin, models.Model):
+
+    class Meta:
+        verbose_name = _('network terminator')
+        verbose_name_plural = _('network terminators')
+        ordering = ('name',)
+
+
+class DiscoveryQueue(NamedMixin, models.Model):
+
+    class Meta:
+        verbose_name = _('discovery queue')
+        verbose_name_plural = _('discovery queues')
+        ordering = ('name',)
+
+
+class IPAddress(LastSeenMixin, TimeStampMixin, models.Model):
+    asset = models.ForeignKey(
+        Asset, verbose_name=_('asset'), null=True, blank=True,
+        default=None, on_delete=models.SET_NULL,
+    )
+    address = models.GenericIPAddressField(
+        _('IP address'), help_text=_('Presented as string.'), unique=True,
+        blank=True, null=True, default=None,
+    )
+    number = models.BigIntegerField(
+        _('IP address'), help_text=_('Presented as int.'), editable=False,
+        unique=True,
+    )
+    # TODO: hostname in asset?
+    # hostname = models.CharField(
+    #     _('hostname'), max_length=255, null=True, blank=True, default=None,
+    # )
+    # hostname.max_length vide /usr/include/bits/posix1_lim.h
+    snmp_name = models.TextField(
```

Comment:
TODO: some of these fields are necessary only in scan (afaik snmp_*, http_family, last_plugins?, is_buried, scan_summary). Maybe we should make some one-to-one model which will have these fields (only for scan purposes) and simplify IPAddress?